lib/LaTeXML/Package/LaTeX.pool.ltxml

```diff
@@ -2710,7 +2710,28 @@ sub beginBibliography {
   setupPseudoBibitem();
   return; }
 
-DefMacro('\bibliography Semiverbatim', '\ifx.#1.\else\@bibliography{#1}\fi');
+DefMacro('\bibliography Semiverbatim', '\lx@ifusebbl{#1}{\input{\jobname.bbl}}{\@bibliography{#1}}');
+DefMacro('\lx@ifusebbl{}{}{}', sub {
+    my ($gullet, $bib_files, $bbl_clause, $bib_clause) = @_;
+    $bib_files = ToString($bib_files);
+
+    if (not $bib_files) {
+      return; }
+
+    my $bbl_path     = FindFile(ToString(Expand(T_CS('\jobname'))) . '.bbl');
```

Comment:
You can pass a ``` type => 'bbl' ``` option into FindFile, which allows you to avoid the concatenation business.